monkey/infection_monkey/exploit/zerologon.py

```diff
@@ -120,16 +134,24 @@ def _send_exploit_rpc_login_requests(self, rpc_con) -> bool:
         return False
 
     def try_exploit_attempt(self, rpc_con) -> Optional[object]:
+        error_message = ""
         try:
             exploit_attempt_result = self.attempt_exploit(rpc_con)
             return exploit_attempt_result
-        except nrpc.DCERPCSessionError as e:
+        except nrpc.DCERPCSessionError as err:
             # Failure should be due to a STATUS_ACCESS_DENIED error.
             # Otherwise, the attack is probably not working.
-            if e.get_error_code() != self.ERROR_CODE_ACCESS_DENIED:
-                logger.info(f"Unexpected error code from DC: {e.get_error_code()}")
-        except BaseException as e:
-            logger.info(f"Unexpected error: {e}")
+            if err.get_error_code() != self.ERROR_CODE_ACCESS_DENIED:
+                error_message = f"Unexpected error code from DC: {err.get_error_code()}"
+                logger.info(error_message)
+        except Exception as err:
+            error_message = f"Unexpected error: {err}"
+            logger.info(error_message)
+
+        self._publish_exploitation_event(
+            success=False,
+            error_message=error_message,
+        )
```

Comment:
This is unnecessary and repetitive in the failing case. We're publishing an exploitation event in `assess_exploit_attempt()` after this.

Reply:
The `assess exploit_attempt` publishes an events if we have any exploit result. The above covers if we don't have any exploitation result for some reason.

Reply:
Let's move this to `assess_exploit_attempt()` for readability, then.

Reply:
I would not like to move it there because in that case, we are losing the error message that happened when we don't have exploit result.